usersession/client/client.go

```diff
@@ -394,7 +394,8 @@ func (client *Client) PendingRefreshNotification(ctx context.Context, refreshInf
 
 // FinishedSnapRefreshInfo holds information about a finished refresh provided to userd.
 type FinishedSnapRefreshInfo struct {
-	InstanceName string `json:"instance-name"`
+	InstanceName    string `json:"instance-name"`
+	AppDesktopEntry string `json:"busy-app-desktop-entry,omitempty"`
```

Comment:
This field may not be needed at all then.

Reply:
Since this part is executed in the user-side daemon, which only receives the instance name, I think that it still is required. Unless it is replaced by SnapName and the .desktop file name building moved into the user-side daemon...

Reply:
Yes, that'd be preferred. My point is that at the place where it's used inside the notification callback there is not enough information to go on to pick the desktop file (remember that a snap can carry many applications, each with a desktop file). AFAIU the purpose of the notification is to just let `snap userd` know that a particular snap was refreshed. One could argue that maybe a revision could be useful, but since it's called from link handler, the `current` symlink should be valid and pointing to the right revision anyway.

Reply:
It seems that the "snap" package isn't accessible from the rest_api.go file...

Reply:
Ok, it is...

Reply:
Done.